R/check-all.R

```diff
@@ -0,0 +1,260 @@
+#' Run all validation checks
+#'
+#' Runs all validation checks. Proper function requires an
+#' environment configuration (config) to be set. The config
+#' is expected to have templates for each metadata_type,
+#' where individual and biospecimen depend on species and
+#' assay depends on the assay type. Additionally, there
+#' should be complete_columns for each metadata_type.
+#'
+#' @param data A tibble or dataframe with the columns:
+#'   file_name, metadata_type, species, assay, file_data.
+#'   The file_data column should be a list column containing
+#'   a dataframe with the file data or `NULL` if the data
+#'   does not exist. `data` is expected to have four rows,
+#'   one for each metadata_type: individual, biospecimen,
+#'   assay, manifest. If file_data is `NULL` for a given
+#'   metadata_type, the metadata_type should still be
+#'   present.
+#' @inheritParams check_annotation_keys
+#' @return List of conditions
+#' @export
+#' @examples
+#' \dontrun{
+#' syn <- synapse$Synapse()
+#' syn$login()
+#'
+#' annots <- get_synapse_annotations(syn = syn)
+#'
+#' data <- tibble::tibble(
+#'   metadata_type = c(
+#'     "manifest",
+#'     "individual",
+#'     "biospecimen",
+#'     "assay"
+#'   ),
+#'   file_name = c("a", NA, NA, "c"),
+#'   species = "human",
+#'   assay = "rnaSeq",
+#'   file_data = c(
+#'     list(data.frame(a = c(TRUE, FALSE), b = c(1, 3))),
+#'     list(NULL),
+#'     list(NULL),
+#'     list(data.frame(a = c(TRUE, FALSE), b = c(1, 3)))
+#'   )
+#' )
+#' res <- check_all(data, annots, syn)
+#' }
+check_all <- function(data, annotations, syn) {
+
+  # Get indices by type
+  indiv_index <- which(data$metadata_type == "individual")
+  biosp_index <- which(data$metadata_type == "biospecimen")
+  assay_index <- which(data$metadata_type == "assay")
+  manifest_index <- which(data$metadata_type == "manifest")
```

Comment:
Can we add a defensive check to make sure that there is only one of each for now? (obviously we'd take it out later if we add support for multiple assays)

Reply:
Added. The check I added has a long error message. I put a `# nolint`, but I remember you mentioning that something complained about our `# nolint`s. Is this going to be an issue?